doc: Updated abi compatibility to ISE +33 (#30044)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/abi_compatibility.rst

The following sections provide detailed lists of changes by component.

IronSide Secure Enclave (IronSide SE) v23.8.0+33
================================================

Added
-----

> * Support for the Nordic nRF global software watchdog timer (GSWDT).

Reviewer: ```suggestion
* Support for the Nordic nRF Global Software Watchdog Timer (GSWDT).
```

>   This can be enabled via a Zephyr driver using the devicetree node ``cpuapp_gswdt``.

Reviewer: ```suggestion
  You can enable this feature through the Zephyr driver using the ``cpuapp_gswdt`` devicetree node.
```
